File: problem3/Problem3.java
import java.lang.Math.*;
import java.util.*;

public class Problem3 {
	public static void main(String[] args) {
		Problem3 p3 = new Problem3();
		p3.problem3();
	}

	public void problem3() {
		long number = 600851475143L;
		//long number = 13195L;
		for(long i=(int)Math.sqrt(number); i>2; i--) {
			if(number%i==0 && isPrime(i)) {
				System.out.println(i);
				return;
			}
		}
	}

	private boolean isPrime(long num) {
		if(num<=3) {
			if(num<=1) {
				return false;
			}
			return true;
		}
		if(num%2==0 || num%3==0) {
			return false;
		}
		for(long i=5; i<(long)(Math.pow(num,0.5)+1); i+=6) {
			if(num%i==0 || (num%(i+2))==0) {
				return false;
			}
		}
		return true;
	}	

}

File: problem3/problem3.py
import math

def is_prime(num):
	if num<=3:
		if num<=1:
			return False
		return True
	if num%2==0 or num%3==0:
		return False
	for i in range(5, int(num**0.5) +1, 6):
		if num%i==0 or (num%(i+2))==0:
			return False
	return True

def problem3():
	number = 600851475143
	#number = 13195
	for i in range((int)(math.sqrt(number)), 1, -1):
		if number%i==0 and is_prime(i):
			print(i)
			return


if __name__=='__main__':
	problem3()
